libs/wingc/src/jsify/codemaker.rs

```diff
@@ -153,26 +318,79 @@ mod tests {
 				<
 				  hello
 				  world
-				>
-			"#}
+				>"#}
 		);
 	}
 
 	#[test]
-	fn codemaker_insert_line() {
+	fn codemaker_append() {
 		let mut code = CodeMaker::default();
-		code.open("if true {");
-		code.line("let b = 2;");
-		code.close("}");
-		code.insert_line(1, "let a = 1;");
-		assert_eq!(
-			code.to_string(),
-			indoc! {r#"
-				if true {
-				  let a = 1;
-				  let b = 2;
-				}
-			"#}
-		);
+		code.append("1");
+		code.append("2");
+		code.append("\n3");
+
+		assert!(code.lines.len() == 2);
+		assert_eq!(code.to_string(), "12\n3");
+	}
+
+	#[test]
+	fn sourcemap_default() {
```

Comment:
What's the behavior if you try to get a sourcemap when there's no span data for any of the lines? Does it produce an empty source map? Let's add a test

Reply:
By default you at least must have the source file, so each line will be mapped to the file (no source line/char). Added test.